### ISSUE TITLE
Typerror rendering Mermaid diagram

### ISSUE THREAD

**Issue by andymarden** [outline/outline#7815]
I have a page with a mermaid diagram embedded. In place of the diagram it says (edit or view mode):

![image](https://github.com/user-attachments/assets/671a9db7-9070-4b85-a90d-f332d4aff794)

If I refresh the page, it then renders OK. Quite a blocker to seamless use! This is on your cloud-hosted implementation

I don't think it is code-dependent (race condition with browser loading?) but here is an example:
```
flowchart TD
    U([web-gui])
    subgraph "xxxxxx Service"
        R["xxxxxx-rest-api"]
        RDB[(xxxxxxxxxxxxxxx)]
        R <--> RDB
    end
    subgraph "xxxxxxxxxxxx"
        A["xxxxxxxx"]
        E[(processed_events.json)]
        T[Templates.json]
        A --> E 
        T --> A
    end
    subgraph "xxxxxx Service"
        CW["xxxxxx-grpc-api server\n + xxxxxx client"]
        D[("xxxxxr Database (TTTTTTTT)")]
        CW -->|"batch:\ntransfers[1..8090]"| D
        D -->|"xxxxxx[]"| CW

    end
    U -->|"read request"| R
    R -->|"lookup xxxxxx"| CW

    subgraph "performance testing"
        X[transfers]
        G[xxxxxxx.py]
        PT["xxxxxxx.py\n + xxxxxxxx client"]
        G --> X --> PT --> D
    end
```

**Comment by tommoor** [comment]
I can't reproduce, though I have seen this error in our logs – could you provide some more information about the reset of the content that's on the page and in which situations the error appears vs doesnt

**Comment by intech** [comment]
I suspect that the error occurs due to cached pages. When we refresh a specific page, the diagram renders correctly; however, when we switch to other pages, the same error reappears.

Additionally, there's another scenario where the diagram appears on one page, but an error shows up on another. After refreshing that second page, the diagram appears correctly there. However, if we then return to the first page (where the diagram was visible before), we find that there is now an error there as well. Refreshing the page resolves the issue.

**Comment by tommoor** [comment]
@andymarden are you still able to reproduce?

**Comment by andymarden** [comment]
Sorry - let me try again.

**Comment by enginewang** [comment]
I also encounter this problem, the same error
